spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/repository/query/SpannerStatementQueryExecutor.java

```diff
@@ -193,16 +216,45 @@ public static Statement buildStatementFromSqlWithArgs(String sql, List<String> t
 		Statement.Builder builder = Statement.newBuilder(sql);
 		for (int i = 0; i < tags.size(); i++) {
 			bindParameter(builder.bind(tags.get(i)), paramStructConvertFunc, spannerCustomConverter,
-					params[i]);
+					params[i], queryMethodParams == null ? null : queryMethodParams.get(tags.get(i)));
 		}
 		return builder.build();
 	}
 
+	private static void bindArrayParameter(ValueBinder<Statement.Builder> bind,
+			SpannerCustomConverter spannerCustomConverter,
+			Iterable originalParam, Parameter paramMetadata) {
+		Iterable param = originalParam;
+		Class innerType = (Class) ((ParameterizedType) paramMetadata.getParameterizedType())
+				.getActualTypeArguments()[0];
+		BiConsumer<ValueBinder<?>, Iterable> toMethod = getIterableValueBinderBiConsumer(innerType);
+
+		// attempt conversion of elements inside
+		if (toMethod == null) {
```

Comment:
should this logic be in `ConverterAwareMappingSpannerEntityWriter`?

Reply:
I tried doing this but I decided against it. I tried extracting a method but the signature needed to have several arguments and they were rather specialized. I felt that it was an uncomfortably specialized method that shouldn't be public on ConverterAwareMappingSpannerEntityWriter

Reply:
@dmitry-s actually great idea. I refactored . PTAL!